cli/azd/pkg/tools/azcli/ad.go

```diff
@@ -350,3 +356,20 @@ func (cli *azCli) createRoleAssignmentsClient(
 
 	return client, nil
 }
+
+func (cli *azCli) CheckRoleAssignments(
```

Comment:
This is not OK.

`cli.getRoleDefinition` checks if a definition exists for some scope.  For example, we use if to check if the role `Owner` exists for a `Resouce Group` or for a `Subscription`.  This method will not tell who this role is assigned to. It will ONLY help to ensure a role can be assigned to someone.

For example, if I want to assign the role `Foo` to a person `Bar`, for the resource `XYZ`, I can first get the role `Foo` from `XYZ`.  And if the role is found, then I can try to assign it to `Bar`.  But if the role is not found within `XYZ`, then there would be an error trying to assign it to someone.

We need `CheckRoleAssignments()` to take a Principal-id and use it to get the role assignments for that principal-id and the resource.  For the previous example, the function would take `Bar` id and `XYZ`.  It will then return all the roles that are assigned to `Bar` for `XYZ`.  Then someone can loop the list and check if `Foo` is part of the list or not